ca/server.go

```diff
@@ -530,13 +534,8 @@ func (s *Server) evaluateAndSignNodeCert(ctx context.Context, node *api.Node) {
 
 // signNodeCert does the bulk of the work for signing a certificate
 func (s *Server) signNodeCert(ctx context.Context, node *api.Node) {
-	if !s.securityConfig.RootCA().CanSign() {
-		log.G(ctx).WithFields(logrus.Fields{
-			"node.id": node.ID,
-			"method":  "(*Server).signNodeCert",
-		}).Errorf("no valid signer found")
-		return
-	}
+	rootCA := s.securityConfig.RootCA()
+	externalCA := s.securityConfig.ExternalCA()
```

Comment:
I originally wanted to use the `MutableTLSCreds` from the `SecurityConfig` so that we could benefit from reusing connections, but after speaking with @diogomonica about it, it turns out that it's not easy to use it with an HTTP client. It would apparently require more changes and a custom `http.RoundTripper` which can renegotiate a TLS handshake when the underlying config changes. We agreed that it is acceptable, for now, to just create a new http client for every request, especially since they don't happen that often.